src/oscar/core/thumbnails.py

```diff
@@ -27,15 +31,18 @@ def generate_thumbnail(self, source, **opts):
         width, height = size.split('x')
         # Set `size` to `width` if `height` is not provided.
         size = size if height else width
-        return get_thumbnail(source, size, **opts)
+        thumbnail = get_thumbnail(source, size, **opts)
```

Comment:
I'm just looking at the code for `get_thumbnail` and can't see a situation where it would return a falsy result? The sorl template tag (which we don't use) expects `get_thumbnail` to throw exceptions and it catches these itself to return an empty string. 

It seems like we may have a bigger problem here which is that thumbnail errors are not caught if `THUMBNAIL_DEBUG` is `False` - think it's worth adding a test for this to make sure we're handling it correctly.

Reply:
> It seems like we may have a bigger problem here which is that thumbnail errors are not caught if THUMBNAIL_DEBUG is False - think it's worth adding a test for this to make sure we're handling it correctly.

What this test would try to reproduce?

Reply:
A test to ensure that if `DEBUG=False`, exceptions raised by the thumbnailer are caught (as sorl's template tag would do by default). Looking at the current code it doesn't seem like they are?

Reply:
Exceptions, raised by the thumbnailer are already handled in the template tag - https://github.com/django-oscar/django-oscar/blob/master/src/oscar/templatetags/image_tags.py#L108-L116:

```python
def render(self, context):
    try:
        return self._render(context)
    except Exception as e:
        if getattr(settings, 'OSCAR_THUMBNAIL_DEBUG', settings.DEBUG):
            raise e

        logger.exception(e)
        return ''
```

Shall we address anything else here @solarissmoke ?